esrally/utils/opts.py

```diff
@@ -17,13 +17,25 @@
 
 import difflib
 import json
+import re
 
 from esrally.utils import io
 
+# detect (very simplistically) json that starts with an array
+RE_JSON_ARRAY_START = re.compile(r'^(\s*\[\s*\]|(\s*\[\s*".*))')
```

Comment:
The parentheses confused me. If we want to keep them, let's use them for each group, like `(...)|(...)`? But we can also leave them out:

```suggestion
# detect (very simplistically) json that starts with an empty array or array of strings
RE_JSON_ARRAY_START = re.compile(r'^\s*\[\s*\]|\s*\[\s*".*')
```

Reply:
> If we want to keep them, let's use them for each group, like `(...)|(...)`?

Yes the intention was to have `(...)|(...)` but the end parens of the first group wrongly ended up at the end up of the expression.

Btw, I contemplated on using `r"""` and having line separated entries, but it felt a bit of an overkill for a regex that is supposed to just check those two things.

I feel the parens add value to emphasize that we are checking two different things, so I'd go for

`r'^(\s*\[\s*\])|(\s*\[\s*".*)'`

Reply:
Sounds good!

Reply:
Fixed in 837bd15ba7890f1ba055324855f08d0014e20e98